namer/namer.cc

```diff
@@ -2084,7 +2154,13 @@ vector<ast::ParsedFile> symbolizeTrees(const core::GlobalState &gs, vector<ast::
         fileq->push(move(tree), 1);
     }
 
-    workers.multiplexJob("symbolizeTrees", [&gs, fileq, resultq, bestEffort]() {
+    // Store a map of common prefix -> file where that prefix is seen.
+    UnorderedMap<core::ClassOrModuleRef, core::FileRef> commonPrefixes;
+
+    // Mutex for safe concurrent updates of map.
+    std::mutex prefixMutex;
```

Comment:
whoa whoa this does not look good. Can we get this working any other way?

putting a mutex here in an otherwise parallel loop is going to throttle our performance and is a code smell

Reply:
Sure, yep, will fix.